src/analyzer/analyzerthread.cpp

```diff
@@ -311,11 +311,12 @@ AnalyzerThread::AnalysisResult AnalyzerThread::analyzeAudioSource(
             const double frameProgress =
                     double(audioSource->frameLength() - remainingFrameRange.length()) /
                     double(audioSource->frameLength());
+            // math_min is required to compensate rounding errors
```

Comment:
This can't be a rounding issue. Because 1 is a exact double value. 
I guess this happens if remainingFrameRange becomes negative. 

How can this ever happen? Did you discover a real bug?

Reply:
```
Debug [AnalyzerThread 0 #1]: audioSource->frameLength() 12126912 remainingFrameRange.length() 2752
Debug [AnalyzerThread 0 #1]: audioSource->frameLength() 12126912 remainingFrameRange.length() 0
Critical [AnalyzerThread 0 #1]: DEBUG ASSERT: "progress <= kAnalyzerProgressFinalizing" in function AnalyzerThread::AnalysisResult AnalyzerThread::analyzeAudioSource(const mixxx::AudioSourcePointer &) at ../src/analyzer/analyzerthread.cpp:321
```

Adding a debug output for either `frameProgress` or `progress` fixes the issue. Register roulette, 80-bit vs. 64-bit float.